login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreen.tsx

```diff
@@ -26,15 +26,18 @@ export const VerifyCodeScreen: React.FC<VerifyCodeScreenProps> = (props) => {
     const [verifyCode, setVerifyCode] = useState(screenData.VerifyCode.code);
     const [isLoading, setIsLoading] = useState(false);
 
-    const requestResendCode = useCallback(async (): Promise<void> => {
-        try {
-            setIsLoading(true);
-            await actions().validateUserRegistrationRequest(verifyCode);
-            setIsLoading(false);
-        } catch {
-            console.error('Error fetching resend verification code!');
-        }
-    }, [verifyCode, actions]);
+    const requestResendCode = useCallback(
+        async (email?: string): Promise<void> => {
+            try {
+                setIsLoading(true);
+                await actions().requestRegistrationCode(email);
+                setIsLoading(false);
+            } catch {
+                console.error('Error fetching resend verification code!');
```

Comment:
Can you move setIsLoading(false) here it will set the loader for both success and error condition?

Reply:
I believe if an error is thrown the catch block executes and then the rest of the code in this block is skipped. Correct me if I'm wrong, but I think you will still be locked in the loading state if an error is throw. You can add a finally block to cover the end of the try and the catch.